### ISSUE TITLE
firebase packages depend on plugin_platform_interface ^2.0.0

### ISSUE THREAD

**Issue by Simpler1** [Baseflow/flutter-permission-handler#464]
## 🐛 Bug Report

flutter_permission_handler 5.1.0 is not compatible with the latest firebase packages.

Each of the latest firebase packages depend on plugin_platform_interface ^2.0.0,
but flutter_permission_handler 5.1.0 depends on plugin_platform_interface ^1.0.2

### Expected behavior
Use flutter_permission_handler with the latest firebase packages.

### Reproduction steps

pubspec.yaml:

```
dependencies:
  cloud_firestore: ^1.0.0
  firebase_auth: ^1.0.0
  firebase_core: ^1.0.0 
  firebase_core_web: ^1.0.0
  firebase_messaging: ^9.0.0
  firebase_storage: ^8.0.0
  permission_handler: ^5.1.0

```
`flutter pub get
`
```
And because permission_handler >=5.1.0 depends on permission_handler_platform_interface ^2.0.2 which depends on plugin_platform_interface ^1.0.2, firebase_storage ^8.0.0 is incompatible with permission_handler >=5.1.0.

And because permission_handler >=5.1.0 depends on permission_handler_platform_interface ^2.0.2 which depends on plugin_platform_interface ^1.0.2, firebase_messaging ^9.0.0 is incompatible with permission_handler >=5.1.0.

And because permission_handler >=5.1.0 depends on permission_handler_platform_interface ^2.0.2 which depends on plugin_platform_interface ^1.0.2, firebase_auth ^1.0.0 is incompatible with permission_handler >=5.1.0.

And because permission_handler >=5.1.0 depends on permission_handler_platform_interface ^2.0.2 which depends on plugin_platform_interface ^1.0.2, firebase_core_web ^1.0.0 is incompatible with permission_handler >=5.1.0.

And because permission_handler >=5.1.0 depends on permission_handler_platform_interface ^2.0.2 which depends on plugin_platform_interface ^1.0.2, firebase_core ^1.0.0 is incompatible with permission_handler >=5.1.0.

And because permission_handler >=5.1.0 depends on permission_handler_platform_interface ^2.0.2 which depends on plugin_platform_interface ^1.0.2, cloud_firestore ^1.0.0 is incompatible with permission_handler >=5.1.0.
```

### Configuration

**Version:** 5.1.0

**Platform:** 
- [x] :iphone: iOS
- [x] :robot: Android

**Comment by MichalDanielDobrzanski** [comment]
This is an important issue.

**Comment by m-j-g** [comment]
Just ran into the same conflict as mentioned.  Flutter 2 released today, and so everyone needs to be able to use the 1.0 firebase packages.

**Comment by mvanbeusekom** [comment]
We are in the progress of fixing this (part of null migration, see #450 and #451). At the moment we are testing the plugin and are expecting a release soon.

**Comment by denkoul** [comment]
A quick hack that i found:
Adding the following, before dev_dependencies
```
dependency_overrides:
  plugin_platform_interface: '>=2.0.0'
```
will force each package that is using eg. `plugin_platform_interface` to use the given version.
It is risky though. As dart website claims:

> Warning: Using a dependency override involves some risk. For example, using an override to specify a version outside the range that the package claims to support, or using an override to specify a local copy of a package that has unexpected behaviors, may break your application.

https://dart.dev/tools/pub/dependencies#dependency-overrides

**Comment by mvanbeusekom** [comment]
Today we released version 6.0.0 of the permission_handler which resolves this issue.
